FIAT/kong_mulder_veldhuizen.py

```diff
@@ -0,0 +1,165 @@
+                2: {0: [i for i in range(12, 18)]},
+            }
+    elif degree == 5:
+        if sd == 2:
+            etop = [[9, 3, 4, 10], [12, 6, 5, 11], [13, 7, 8, 14]]
+            entity_ids = {
+                0: dict((i, [i]) for i in range(3)),
+                1: dict((i, etop[i]) for i in range(3)),
+                2: {0: [i for i in range(15, 30)]},
+            }
+    return entity_ids
+
+
+def bump(T, deg):
+    """Increase degree of polynomial along"""
+    sd = T.get_spatial_dimension()
+    if deg == 1:
+        return (0,)
+    if sd == 2:
+        if deg < 5:
+            return (1,)
+        elif deg == 5:
+            return (2,)
+    elif sd == 3:
+        return (1, 2)
+
+
+def KongMulderVeldhuizenSpace(T, deg):
+    sd = T.get_spatial_dimension()
+    if deg == 1:
+        return Lagrange(T, 1).poly_set
+    else:
+        L = Lagrange(T, deg)
+        # Toss the bubble from Lagrange since it's dependent
+        # on the higher-dimensional bubbles
+        if sd == 2:
+            inds = [
+                i
+                for i in range(L.space_dimension())
+                if i not in L.dual.entity_ids[sd][0]
+            ]
+        elif sd == 3:
+            not_inds = [L.dual.entity_ids[sd][0]] + [
+                L.dual.entity_ids[sd - 1][f] for f in L.dual.entity_ids[sd - 1]
+            ]
+            not_inds = [item for sublist in not_inds for item in sublist]
+            inds = [i for i in range(L.space_dimension()) if i not in not_inds]
+        RL = RestrictedElement(L, inds)
+        # interior cell bubble
+        bubs = Bubble(T, deg + max(bump(T, deg)))
+        if sd == 2:
+            return NodalEnrichedElement(RL, bubs).poly_set
+        elif sd == 3:
+            # bubble on the facet
+            fbubs = FacetBubble(T, deg + min(bump(T, deg)))
+            return NodalEnrichedElement(RL, bubs, fbubs).poly_set
+
+
+class KongMulderVeldhuizenDualSet(dual_set.DualSet):
+    """The dual basis for KMV simplical elements."""
+
+    def __init__(self, ref_el, degree):
+        entity_ids = {}
+        entity_ids = _get_topology(ref_el, degree)
+        lr = create_quadrature(ref_el, degree, scheme="KMV")
+        nodes = [functional.PointEvaluation(ref_el, x) for x in lr.pts]
+        super(KongMulderVeldhuizenDualSet, self).__init__(nodes, ref_el, entity_ids)
+
+
+class KongMulderVeldhuizen(finite_element.CiarletElement):
+    """The "lumped" simplical finite element (NB: requires custom quad. "KMV" points to achieve a diagonal mass matrix).
+
+    References
+    ----------
+
+    Higher-order triangular and tetrahedral finite elements with mass
+    lumping for solving the wave equation
+    M. J. S. CHIN-JOE-KONG, W. A. MULDER and M. VAN VELDHUIZEN
+
+    HIGHER-ORDER MASS-LUMPED FINITE ELEMENTS FOR THE WAVE EQUATION
+    W.A. MULDER
+
+    NEW HIGHER-ORDER MASS-LUMPED TETRAHEDRAL ELEMENTS
+    S. GEEVERS, W.A. MULDER, AND J.J.W. VAN DER VEGT
+
+     """
```

Comment:
FIAT doesn't hook into the PETSc citations mechanism like Firedrake & FIAT do, but this would be useful information to include in the FInAT wrapper to this element.

Reply:
Would you be okay with leaving this in? I will make a note to put them in FInAT too.

Reply:
Yes.  Good to leave it in the FIAT source, but going into FInAT via the `Citations` business would be nice.